src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -649,3 +645,30 @@ class DelayedResizer extends Disposable {
 		this._register(toDisposable(() => clearTimeout(this._timeout)));
 	}
 }
+
+/**
+ * Splits incoming pty data into chunks to try prevent data corruption that could occur when pasting
+ * large amounts of data.
+ */
+export function chunkInput(data: string): string[] {
+	const chunks: string[] = [];
+	let nextChunkStartIndex = 0;
+	for (let i = 0; i < data.length - 1; i++) {
+		if (
+			// If the max chunk size is reached
+			i - nextChunkStartIndex + 1 >= Constants.WriteMaxChunkSize ||
+			// If the next character is ESC, send the pending data to avoid splitting the escape
+			// sequence.
+			data[i + 1] === '\x1b'
+		) {
+			chunks.push(data.substring(nextChunkStartIndex, i + 1));
+			nextChunkStartIndex = i + 1;
+			i++;
```

Comment:
Potentially confusing with i++ both here and in the for loop itself. Maybe it can be a while loop?

Reply:
I'll leave a comment, that just skips the next character as it'd be 1 in length to save a little time